Northstar.Client/mod/scripts/vscripts/ui/menu_mod_settings.nut

```diff
@@ -0,0 +1,1064 @@
+untyped
+global function AddModSettingsMenu
+global function AddConVarSetting
+global function AddConVarSettingEnum
+global function AddConVarSettingSlider
+global function AddModSettingsButton
+global function AddModTitle
+global function AddModCategory
+global function PureModulo
```

Comment:
Just like Server dialog and Server RUI these are custom functions so I'd prefix them with `NS`

Reply:
Not breaking compatibility again, thank you.

Reply:
That's just adding unnecessary mess. In addition, if we would be keeping the original function signature, there is no need for the alias. There's no reason to change it.

Reply:
> Not breaking compatibility again, thank you.

It doesn't matter, this PR will break compatibility for a decent number of mods anyway, since they use dependency constants, which won't be set since this is no longer a separate mod

Reply:
The `NS` prefix is supposed to signify that these are non vanilla functions. Even if it bears wery little value I still think it should be added.

On the topic of function names I feel like they should all have smth like `ModSettings` in their name as for example `AddConVarSetting` could mislead ppl into thinking this adds an entry to the vanilla settings menu.

Reply:
So what you propose is to make an alias for each function respectively like `NSModSettingsAddConVarSetting`? Doesn't sound very nice to type

Reply:
It may not be very nice to look at, but it tells you all you need to know about what the function does purely based on its name

Reply:
But it already does:

look at this code and tell me what it does:

```nut
AddModTitle("HUD ^FF902000Revamp")
AddModCategory("General")
AddConVarSettingEnum("comp_hud_layout", "Layout", [ "HUDRevamp", "Destiny 2" ] )
AddModCategory("Damage Indicator")
AddConVarSettingEnum( "comp_hud_damage_indicator", "Enabled", [ "No", "Yes" ] )
AddConVarSetting( "comp_hud_damage_indicator_duration", "Duration", "float" )
AddConVarSetting( "comp_hud_damage_indicator_fade_time", "Fade Time", "float" )

AddModCategory("Incoming Damage Indicator")
AddConVarSettingEnum( "comp_hud_incoming_damage_indicator", "Enabled", [ "No", "Yes" ] )
AddConVarSetting( "comp_hud_incoming_damage_indicator_duration", "Duration", "float" )
AddConVarSetting( "comp_hud_incoming_damage_indicator_fade_time", "Fade Time", "float" )
```

tell me what it does. There is ***no way***, this will be confused for something else, unless the modder uses some stupid proprietary system for a settings menu (which is *their problem*, not ours.)

Reply:
I can add aliases but I would like to keep compatibility as it is right now